internal/cli/cmd/login/login.go

```diff
@@ -177,13 +181,29 @@ func (o *option) authAndSetToken() error {
 			return ctx.Err()
 		case <-ticker.C:
 			tokenResponse, tokenErr := o.apiClient.Auth().DeviceAccessToken(string(o.externalAuthKind), o.externalAuthProvider, deviceCodeResponse.DeviceCode)
-			if tokenErr != nil {
-				continue
+			if tokenErr == nil {
+				// Handle the access token
+				o.accessToken = tokenResponse.AccessToken
+				o.refreshToken = tokenResponse.RefreshToken
+				return nil
+			}
+
+			reqErr := &perseshttp.RequestError{}
+			if errors.As(tokenErr, &reqErr) && reqErr.OAuthError != nil {
```

Comment:
I think you are missing the case when `tokenErr` is not a `perseshttp.RequestError`

Reply:
and the case when `reqErr.OauthErro == nil`

Reply:
updated